src/analysis/normed_space/basic.lean

```diff
@@ -1180,6 +1181,16 @@ instance prod.semi_normed_ring [semi_normed_ring β] : semi_normed_ring (α × 
         ... = (∥x∥*∥y∥) : rfl,
   ..prod.semi_normed_group }
 
+/-- seminormed group instance on a matrix with values in a seminormed group,
+using the sup norm -/
+instance {n m : Type*} [fintype n] [fintype m] : semi_normed_group (matrix n m α) :=
+pi.semi_normed_group
```

Comment:
What about the operator norm? (Seems more canonical!) Could this just be a `def`?

What is it for? Perhaps it doesn't belong in this file but closer to the application?

Reply:
Maybe this should be in a locale?

Reply:
I'm going to prove a fact about the topology on the space of matrices (which _is_ canonical), namely that a certain map from one spaces to another is a closed embedding.  But the proof technique is to use `linear_equiv.closed_embedding_of_injective` (see later in this same PR), which requires some normed structure on matrices as an input -- the point is that the topology is compatible with a norm, doesn't really matter what norm (but if I use the operator norm, the compatibility with the topology would probably be harder to prove).

@semorrison @eric-wieser if you prefer I can make all three of these defs rather than instances, and use `local instance ...` at the point of use.

Reply:
FWIW, I vote for the `local instance` approach.

Reply:
I agree, local instance seems perfect to "localize" to a proof technique.